### PR TITLE
Update the permissions migration to look at role names

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Improves permissions migration to work better in production by looking at the role codes. Technically role codes can be anything, but these were the ones during the migration in production, which is the most critical part.

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- None

### CODE REVIEW

File: permissions/migrations/0029_migrate_permissions_to_new_models.py [hunk 1]
```diff
@@ -88,14 +94,16 @@ def migrate_permissions_to_new_models(apps, schema_editor):
...
-def _determine_role(permissions: list[str]) -> str:
-    if permissions == RESERVER_PERMISSIONS:
+def _determine_role(permissions: list[str], role_name: str) -> str:
+    # Role names can technically be anything, but these names are used in production,
+    # which is the most critical part for the migration.
+    if role_name in ["Varaaja", "viewer"] or permissions == RESERVER_PERMISSIONS:
         return UserRoleChoices.RESERVER.value
-    if permissions == VIEWER_PERMISSIONS:
+    if role_name == "Katselija" or permissions == VIEWER_PERMISSIONS:
         return UserRoleChoices.VIEWER.value
```
Comment (ranta) [inline permissions/migrations/0029_migrate_permissions_to_new_models.py:103]: Shouldn't `"viewer"` be mapped to `UserRoleChoices.VIEWER`? 🤔

File: permissions/migrations/0029_migrate_permissions_to_new_models.py [hunk 2]
```diff
@@ -88,14 +94,16 @@ def migrate_permissions_to_new_models(apps, schema_editor):
     NewUnitRoleUnitGroup.objects.bulk_create(new_unit_roles_unit_groups)
 
 
-def _determine_role(permissions: list[str]) -> str:
-    if permissions == RESERVER_PERMISSIONS:
+def _determine_role(permissions: list[str], role_name: str) -> str:
+    # Role names can technically be anything, but these names are used in production,
+    # which is the most critical part for the migration.
+    if role_name in ["Varaaja", "viewer"] or permissions == RESERVER_PERMISSIONS:
```
Comment (vergama) [inline permissions/migrations/0029_migrate_permissions_to_new_models.py:100]: `"viewer"` to `UserRoleChoices.VIEWER` with `"Katselija"`

File: permissions/migrations/0029_migrate_permissions_to_new_models.py (hunk 1 above)
Comment (matti-lamppu) [reply permissions/migrations/0029_migrate_permissions_to_new_models.py]: Whoops. FIxed.

File: permissions/migrations/0029_migrate_permissions_to_new_models.py (hunk 2 above)
Comment (matti-lamppu) [reply permissions/migrations/0029_migrate_permissions_to_new_models.py]: Whoops. FIxed.

**Review by ranta — APPROVED** [review]

**Pull request merged**